pkg/hydrate/clean.go

```diff
@@ -47,7 +47,7 @@ func clean(o ast.FileObject) {
 	// Remove all the annotations starting with configsync.gke.io or configmanagement.gke.io
 	// except for the configmanagement.gke.io/managed annotation.
 	for k := range annotations {
-		if metadata.HasConfigSyncPrefix(k) && k != metadata.ResourceManagementKey {
+		if metadata.HasConfigSyncPrefix(k) && k != metadata.ManagementModeAnnotationKey {
```

Comment:
It seems like this could probably be abstracted away to a helper function which checks whether the annotation is user configurable. My interpretation is that this should skip deleting any Config Sync annotations that are set by the user.

Even if today the only user-configurable CS annotation is `ManagementModeAnnotationKey`, it would be less of a future gotcha if we had this abstracted to a separate `metadata` function IMO

Reply:
Honestly, I don't know why this is here. Seems like invalid fields should cause an error, not silently strip them.

I don't really want to touch this in this PR.